### PR TITLE
[ASP-3906] Add constraints to schemas

### PR DESCRIPTION
**Pull request by julianaklulo**
#### What
Update the API schemas to include constraints in all numerical values, also add max lenght to string values.

#### Why
To ensure the configurations will be created with reasonable values.

`Task`: https://jira.scania.com/browse/ASP-3906

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: backend/lm_backend/api/schemas/booking.py [hunk 1]
```diff
@@ -11,30 +11,40 @@ class BookingCreateSchema(BaseCreateSchema):
     Represents the booking of a feature.
     """
 
-    job_id: int
-    feature_id: int
-    quantity: int
+    job_id: int = Field(..., title="Job ID", description="The ID of the job that booked the feature.")
+    feature_id: int = Field(..., title="Feature ID", description="The ID of the feature that was booked.")
+    quantity: PositiveInt = Field(
```
Comment (dusktreader) [inline backend/lm_backend/api/schemas/booking.py:16]: Should there be an upper bound on this one?

File: backend/lm_backend/api/schemas/license_server.py [hunk 2]
```diff
@@ -23,39 +27,47 @@ class LicenseServerWithoutConfigIdCreateSchema(BaseCreateSchema):
     The config_id will be handled in the endpoint.
     """
 
-    host: str
-    port: PositiveInt
+    host: str = Field(..., title="Host", description="The host of the license server.")
+    port: PositiveInt = Field(..., title="Port", description="The port of the license server.")
```
Comment (dusktreader) [inline backend/lm_backend/api/schemas/license_server.py:45]: This definitely should have an upper bound (65535).

File: backend/lm_backend/api/schemas/license_server.py [hunk 3]
```diff
@@ -23,39 +27,47 @@ class LicenseServerWithoutConfigIdCreateSchema(BaseCreateSchema):
     The config_id will be handled in the endpoint.
     """
 
-    host: str
-    port: PositiveInt
+    host: str = Field(..., title="Host", description="The host of the license server.")
```
Comment (dusktreader) [inline backend/lm_backend/api/schemas/license_server.py:44]: We should also make sure valid hostnames are supplied here. According to https://www.rfc-editor.org/rfc/rfc1123: [a-z0-9-.]+ (basically)

**Review by dusktreader — CHANGES_REQUESTED** [review]
Some small nitpicky ones for you.

File: backend/lm_backend/api/schemas/license_server.py (hunk 3 above)
Comment (julianaklulo) [reply backend/lm_backend/api/schemas/license_server.py:44]: > "Host software MUST handle host names of up to 63 characters and SHOULD handle host names of up to 255 characters."

This regex would consider the 63 characters limit: `"^(?!-)[A-Za-z0-9-]{1,63}(?<!-)(\.[A-Za-z0-9-]{1,63})*$"`

File: backend/lm_backend/api/schemas/booking.py (hunk 1 above)
Comment (julianaklulo) [reply backend/lm_backend/api/schemas/booking.py:16]: The quantity is validaded in the CRUD create booking method.
The method calculates the quantity of licenses free to decide if the booking can be created.
I don't think this is something that can be handled in the schema.

File: backend/lm_backend/api/schemas/license_server.py (hunk 2 above)
Comment (julianaklulo) [reply backend/lm_backend/api/schemas/license_server.py:45]: Added a validator for it.

**Review by dusktreader — APPROVED** [review]

**Pull request merged**